src/AdyenCheckoutContext.tsx

```diff
@@ -142,76 +165,100 @@ const AdyenCheckout: React.FC<AdyenCheckoutProps> = ({
         subscriptions.current.push(
-          eventEmitter.addListener(Event.onAdditionalDetails, (data) =>
-            onAdditionalDetails?.(data, nativeComponent)
-          )
+          eventEmitter.addListener(
+            Event.onAdditionalDetails,
+            (data) => onAdditionalDetails?.(data, nativeComponent),
+          ),
         );
       }
 
       if (nativeComponent.events.includes(Event.onComplete)) {
         subscriptions.current.push(
-          eventEmitter.addListener(Event.onComplete, () =>
-            onComplete?.(nativeComponent)
-          )
+          eventEmitter.addListener(
+            Event.onComplete,
+            (data) => onComplete?.(data, nativeComponent), // TODO: provide result for voucher
+          ),
         );
       }
     },
-    [submitPayment, onAdditionalDetails, onComplete, onError, subscriptions]
+    [submitPayment, onAdditionalDetails, onComplete, onError, subscriptions],
   );
 
   const start = useCallback(
     (typeName: string) => {
       removeEventListeners();
       analytics.current = new Analytics(config);
-      const { nativeComponent, paymentMethod } = getNativeComponent(
+
+      const currentPaymentMethods = checkPaymentMethodsResponse(
+        paymentMethods ?? sessionStorage?.paymentMethods,
+      );
+
+      const {nativeComponent, paymentMethod} = getNativeComponent(
         typeName,
-        paymentMethods
+        currentPaymentMethods,
       );
 
-      checkPaymentMethodsResponse(paymentMethods);
       checkConfiguration(config);
 
       startEventListeners(config, nativeComponent);
 
       if (paymentMethod) {
-        const singlePaymentMethods = { paymentMethods: [paymentMethod] };
+        const singlePaymentMethods = {paymentMethods: [paymentMethod]};
         const singlePaymentConfig = {
           ...config,
-          dropin: { skipListWhenSinglePaymentMethod: true },
+          dropin: {skipListWhenSinglePaymentMethod: true},
         };
-        analytics.current.send({ component: typeName });
+        analytics.current.send({component: typeName});
         nativeComponent.open(singlePaymentMethods, singlePaymentConfig);
       } else {
         analytics.current.send({
           paymentMethods: paymentMethods?.paymentMethods.map((e) => e.type),
           component: 'dropin',
           flavor: 'dropin',
         });
-        nativeComponent.open(paymentMethods, config);
+        nativeComponent.open(currentPaymentMethods, config);
       }
     },
     [
       config,
       paymentMethods,
+      sessionStorage,
       startEventListeners,
       removeEventListeners,
       analytics,
-    ]
+    ],
   );
 
+  const createSession = useCallback(async () => {
+    try {
+      const sessionResponse = await SessionHelper.createSession(
+        session,
+        config,
+      );
+      setSession(sessionResponse);
+    } catch (error) {
+      console.error('Setting up session: ' + error);
```

Comment:
Maybe we can call the `onError` callback, to make it consistent with the Web sdk.

Reply:
Good point! Fixed [dd1158b](https://github.com/Adyen/adyen-react-native/pull/331/commits/dd1158ba37dcdc392b2fa1bece0a267c5e8a6ecb)